src/mlpack/tests/neat_test.cpp

```diff
@@ -0,0 +1,77 @@
+/**
+ * @file neat_test.cpp
+ * @author Rahul Ganesh Prabhu
+ *
+ * Tests for NeuroEvolution of Augmenting Topologies.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+
+#include <mlpack/core.hpp>
+
+// NEAT resources.
+#include <mlpack/methods/neat/neat.hpp>
+#include <mlpack/methods/neat/selection_strategies/rank_selection.hpp>
+
+#include <mlpack/tests/neat_test_tools.hpp>
+
+#include <boost/test/unit_test.hpp>
+#include "test_tools.hpp"
+
+using namespace mlpack;
+using namespace mlpack::rl;
+using namespace mlpack::ann;
+using namespace mlpack::neat;
+
+BOOST_AUTO_TEST_SUITE(NEATTest)
+
+/**
+ * Test NEAT and phased searching on the XOR Test. XOR normally performs better
+ * in cyclic cases, this is meant to be a test of the acyclic network.
+ */
+BOOST_AUTO_TEST_CASE(NEATXORTest)
+{
+  arma::arma_rng::set_seed_random();
```

Comment:
Can you remove the random seed here, for debugging it would be great to have the same random seed for each run.